### PR TITLE
reduce PMD/spotbugs lints in tests

### PR DESCRIPTION
**Pull request by jomiklos**
reduced bundle of lints/warnings reported by PMD in unit/integration tests

### CODE REVIEW

File: src/test/java/com/tailf/jnc/framing/FramingTest.java [hunk 1]
```diff
@@ -144,11 +144,18 @@ void partialChunksTest(String frame, String chunkedFrame) throws IOException {
...
-            assertEquals("should restore the frame from two parts", frame, framer.parseFrame());
+        try {
+            for (int i = 2; i < data.length - 1; i++) {
+                DataReader rdr = new TestDataReader(po, data, i);
+                Framer framer = Framing.CHUNKED.newSessionFramer(rdr, pi,
+                                                                 nullOutputStream());
+                assertEquals("should restore the frame from two parts", frame, framer.parseFrame());
+            }
+        } catch (IOException ex) {
+            throw ex;
```
Comment (martin-volf) [inline src/test/java/com/tailf/jnc/framing/FramingTest.java:155]: Why do you catch the exception then?

File: src/test/java/com/tailf/jnc/framing/FramingTest.java [hunk 2]
```diff
@@ -144,11 +144,18 @@ void partialChunksTest(String frame, String chunkedFrame) throws IOException {
...
+        } finally {
+            po.close();
+            pi.close();
```
Comment (martin-volf) [inline src/test/java/com/tailf/jnc/framing/FramingTest.java:158]: Can't this be rewritten using the auto-closing pattern?

File: tests/1-test-union/src/app/Client.java [hunk 3]
```diff
@@ -4,24 +4,23 @@
 
 import union.C;
 import union.Union;
-import union.c.L;
+// import union.c.L;
```
Comment (martin-volf) [inline tests/1-test-union/src/app/Client.java:7]: Please remove it, if not needed.

File: tests/1-test-union/src/app/Client.java [hunk 4]
```diff
@@ -95,26 +93,26 @@ public static void main(String[] args) throws IOException, JNCException {
...
 
         String configAsXML = cConfig.toXMLString();
         System.out.println("Initial config:\n" + configAsXML);
         System.out.println("HashCode of the config XML string: " +
                 configAsXML.hashCode());
-        
+
 //        // Increment number of servers at each l by one
-        ElementChildrenIterator lIterator = cConfig.lIterator();
+//        ElementChildrenIterator lIterator = cConfig.lIterator();
```
Comment (martin-volf) [inline tests/1-test-union/src/app/Client.java:109]: Remove if not needed; also, I think the surrounding commented-out code can be removed, even if the linter is ok with it.

File: tests/4-test-leafref/src/app/Client.java [hunk 5]
```diff
@@ -97,40 +91,40 @@ public static void main(String[] args) throws IOException, JNCException {
...
         ElementChildrenIterator lIterator = cConfig.l1Iterator();
         while (lIterator.hasNext()) {
             L1 h = (L1) lIterator.next();
-            
+
             // Internal representations of key1 and key2
             YangString key1Value = h.getKey1Value();
-            YangString key2Value = h.getKey2Value();
-            
+            // YangString key2Value = h.getKey2Value();
```
Comment (martin-volf) [inline tests/4-test-leafref/src/app/Client.java:112]: remove

File: src/test/java/com/tailf/jnc/YangTypeTest.java [hunk 6]
```diff
@@ -17,17 +16,21 @@ public class YangTypeTest {
...
+        @SuppressWarnings("unchecked") @Override
         protected T fromString(String s) { return (T)s; }
-        @Override public void check() throws YangException {}
-        @Override public boolean canEqual(Object obj) { 
-            return value == null ? false : obj instanceof YangTypeDummy;
+        @Override public void check() throws YangException {
+            // no checks
+        }
+        @Override public boolean canEqual(Object obj) {
+            return value != null || obj instanceof YangTypeDummy;
```
Comment (micnovak) [inline src/test/java/com/tailf/jnc/YangTypeTest.java:33]: Incorrect refactoring ?

**Review by micnovak — APPROVED** [review]

File: src/test/java/com/tailf/jnc/framing/FramingTest.java (hunk 1 above)
Comment (jomiklos) [reply src/test/java/com/tailf/jnc/framing/FramingTest.java:155]: correct, removed...

File: src/test/java/com/tailf/jnc/YangTypeTest.java (hunk 6 above)
Comment (jomiklos) [reply src/test/java/com/tailf/jnc/YangTypeTest.java:33]: you're right! fixed now to `&&`

**Comment by jojohans** [comment]
Any idea why YangBaseStringTest.java and YangBinaryTest.java is marked as every line changed?

**Review by jojohans — APPROVED** [review]

File: tests/1-test-union/src/app/Client.java (hunk 4 above)
Comment (jomiklos) [reply tests/1-test-union/src/app/Client.java]: rolled back the change - left for later cleanup/code enhancement...

File: tests/4-test-leafref/src/app/Client.java (hunk 5 above)
Comment (jomiklos) [reply tests/4-test-leafref/src/app/Client.java]: rolled back the change - left for later cleanup/code enhancement...

File: tests/1-test-union/src/app/Client.java (hunk 3 above)
Comment (jomiklos) [reply tests/1-test-union/src/app/Client.java:7]: rolled back the change - left for later cleanup/code enhancement...

File: src/test/java/com/tailf/jnc/framing/FramingTest.java [hunk 7]
```diff
@@ -168,14 +171,21 @@ public void twoSeparatedFrames() throws IOException {
...
+                Framer framer = Framing.END_OF_MESSAGE.newSessionFramer(rdr, pi,
+                nullOutputStream());
+                assertEquals("should restore both frames from parts",
+                INTL_FRAME + LONGER_FRAME,
+                framer.parseFrame() + framer.parseFrame());
+            }
+        } catch (IOException ex) {
+            throw ex;
+        } finally {
+            pi.close();
```
Comment (martin-volf) [inline src/test/java/com/tailf/jnc/framing/FramingTest.java:187]: Can you use the resource auto-closing here too? Sorry for not notifying you before.

File: src/test/java/com/tailf/jnc/framing/FramingTest.java [hunk 8]
```diff
@@ -184,13 +194,20 @@ public void twoChunkedFrames() throws IOException {
...
+                Framer framer = Framing.CHUNKED.newSessionFramer(rdr, pi,
+                nullOutputStream());
+                assertEquals("should restore both frames from parts",
+                INTL_FRAME + LONGER_FRAME,
+                framer.parseFrame() + framer.parseFrame());
+            }
+        } catch (IOException ex) {
+            throw ex;
+        } finally {
+            pi.close();
```
Comment (martin-volf) [inline src/test/java/com/tailf/jnc/framing/FramingTest.java:209]: And here...

**Comment by jomiklos** [comment]
> Any idea why YangBaseStringTest.java and YangBinaryTest.java is marked as every line changed?

EOL changed from windows format to match `LF` only as rest of the files...

File: src/test/java/com/tailf/jnc/framing/FramingTest.java (hunk 7 above)
Comment (jomiklos) [reply src/test/java/com/tailf/jnc/framing/FramingTest.java]: fixed

File: src/test/java/com/tailf/jnc/framing/FramingTest.java (hunk 8 above)
Comment (jomiklos) [reply src/test/java/com/tailf/jnc/framing/FramingTest.java]: fixed

**Pull request merged**